pinot-core/src/test/java/com/linkedin/pinot/core/startree/hll/TestHllIndexSize.java

```diff
@@ -0,0 +1,187 @@
+                  "column17", "column18");
+  private static final String AVRO_DATA = "data/test_data-sv.avro";
+  private static final String timeColumnName = "daysSinceEpoch";
+  private static final TimeUnit timeUnit = TimeUnit.DAYS;
+
+  private static final int hllLog2m = 5;
+
+  private IndexLoadingConfigMetadata v1LoadingConfig;
+  private IndexLoadingConfigMetadata v3LoadingConfig;
+
+  private HllConfig hllConfig;
+
+  @BeforeMethod
+  public void setUp() throws Exception {
+    hllConfig = new HllConfig(columnsToDeriveHllFields, hllLog2m, hllDeriveColumnSuffix);
+
+    Configuration tableConfig = new PropertiesConfiguration();
+    tableConfig.addProperty(IndexLoadingConfigMetadata.KEY_OF_SEGMENT_FORMAT_VERSION, "v1");
+    v1LoadingConfig = new IndexLoadingConfigMetadata(tableConfig);
+
+    tableConfig.clear();
+    tableConfig.addProperty(IndexLoadingConfigMetadata.KEY_OF_SEGMENT_FORMAT_VERSION,  "v3");
+    v3LoadingConfig = new IndexLoadingConfigMetadata(tableConfig);
+  }
+
+  @AfterMethod
+  public void tearDown() throws Exception {}
+
+  @Test
+  public void testColumnStatsWithoutStarTree() {
+    SegmentWithHllIndexCreateHelper helper = null;
+    boolean hasException = false;
+    try {
+      LOGGER.info("================ Without StarTree ================");
+      helper = new SegmentWithHllIndexCreateHelper(
+              "noStarTree", AVRO_DATA, timeColumnName, timeUnit);
+      SegmentIndexCreationDriver driver = helper.build(false, null);
+      LOGGER.info("================ Cardinality ================");
+      for (String name : helper.getSchema().getColumnNames()) {
+        LOGGER.info("* " + name + ": " + driver.getColumnStatisticsCollector(name).getCardinality());
+      }
+    } catch (Exception e) {
+      hasException = true;
+      LOGGER.error(e.getMessage());
+    } finally {
+      if (helper != null) {
+        helper.cleanTempDir();
+      }
+      Assert.assertEquals(hasException, false);
+    }
+  }
+
+  @Test
+  public void testColumnStatsWithStarTree() {
+    SegmentWithHllIndexCreateHelper helper = null;
+    boolean hasException = false;
+    try {
+      LOGGER.info("================ With StarTree ================");
+      helper = new SegmentWithHllIndexCreateHelper(
+              "withStarTree", AVRO_DATA, timeColumnName, timeUnit);
+      SegmentIndexCreationDriver driver = helper.build(true, hllConfig);
+      LOGGER.info("================ Cardinality ================");
+      for (String name : helper.getSchema().getColumnNames()) {
+        LOGGER.info("* " + name + ": " + driver.getColumnStatisticsCollector(name).getCardinality());
+      }
+    } catch (Exception e) {
+      hasException = true;
+      LOGGER.error(e.getMessage());
+    } finally {
+      if (helper != null) {
+        helper.cleanTempDir();
+      }
+      Assert.assertEquals(hasException, false);
+    }
+  }
+
+  @Test
+  public void testConvert() throws Exception {
```

Comment:
what is the purpose of this test ? It seems to test v1 -> v3 format conversion. Is that relevant to your patch ?